internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -745,87 +749,103 @@ func (s Service) AuthorizeSession(ctx context.Context, req *pbs.AuthorizeSession
-			pluginHostSetIds = append(pluginHostSetIds, hsId)
-		}
-	}
-	if len(pluginHostSetIds) > 0 {
-		eps, err := pluginHostRepo.Endpoints(ctx, pluginHostSetIds)
-		if err != nil {
-			return nil, err
 		}
-		endpoints = append(endpoints, eps...)
-	}
 
-	if len(endpoints) == 0 && t.GetAddress() == "" {
-		return nil, handlers.NotFoundErrorf("No host sources or address found for given target.")
-	}
+		if len(endpoints) == 0 {
+			return nil, handlers.NotFoundErrorf("No host sources or address found for given target.")
+		}
 
-	var chosenEndpoint *host.Endpoint
-	if requestedId != "" {
-		for _, ep := range endpoints {
-			if ep.HostId == requestedId {
-				chosenEndpoint = ep
+		var chosenEndpoint *host.Endpoint
+		if requestedId != "" {
+			for _, ep := range endpoints {
+				if ep.HostId == requestedId {
+					chosenEndpoint = ep
+				}
+			}
+			if chosenEndpoint == nil {
+				// We didn't find it
+				return nil, handlers.InvalidArgumentErrorf(
+					"Errors in provided fields.",
+					map[string]string{
+						"host_id": "The requested host id is not available.",
+					})
 			}
 		}
+
 		if chosenEndpoint == nil {
-			// We didn't find it
-			return nil, handlers.InvalidArgumentErrorf(
-				"Errors in provided fields.",
-				map[string]string{
-					"host_id": "The requested host id is not available.",
-				})
+			chosenEndpoint = endpoints[rand.Intn(len(endpoints))]
 		}
-	}
-
-	var h, p, hostId, hostSetId string
-	if chosenEndpoint == nil && len(endpoints) > 0 {
-		chosenEndpoint = endpoints[rand.Intn(len(endpoints))]
-	}
 
-	if chosenEndpoint != nil {
 		hostId = chosenEndpoint.HostId
 		hostSetId = chosenEndpoint.SetId
-		h, p, err = net.SplitHostPort(chosenEndpoint.Address)
-		switch {
-		case err != nil && strings.Contains(err.Error(), missingPortErrStr):
-			if t.GetDefaultPort() == 0 {
-				return nil, errors.New(ctx, errors.InvalidParameter, op, fmt.Sprintf("neither the selected host %q nor the target provides a port to use", chosenEndpoint.HostId))
-			}
-			h = chosenEndpoint.Address
-			p = strconv.FormatUint(uint64(t.GetDefaultPort()), 10)
-		case err != nil:
-			return nil, errors.Wrap(ctx, err, op, errors.WithMsg("error when parsing the chosen endpoints host address"))
-		}
+		h = chosenEndpoint.Address
 	}
 
-	if t.GetAddress() != "" && hostId == "" && hostSetId == "" {
-		h = t.GetAddress()
-		p = strconv.FormatUint(uint64(t.GetDefaultPort()), 10)
+	if h == "" {
+		return nil, handlers.ApiErrorWithCodeAndMessage(
+			codes.FailedPrecondition,
+			"No host was discovered after checking target address and host sources.")
+	}
+
+	// Ensure we don't have a port from the address, which would be unexpected
+	// FIXME: We've decided to hold off on making this an error until 0.14. In
+	// the meantime, ignore any port coming from the host address.
+	hostWithoutPort, _, err := net.SplitHostPort(h)
+	switch {
+	case err != nil && strings.Contains(err.Error(), missingPortErrStr):
+		// This is what we expect
+		_ = hostWithoutPort
```

Comment:
What is the meaning of this assignment? I think we can remove it

Reply:
Did you mean to leave this around?

Reply:
Nope, thanks!